app/src/sharedTest/java/org/oppia/android/app/onboarding/OnboardingProfileTypeFragmentTest.kt

```diff
@@ -0,0 +1,420 @@
+        .check(
+          matches(
+            hasDescendant(
+              withText(R.string.onboarding_profile_type_activity_parent_text)
+            )
+          )
+        )
+      onView(withId(R.id.onboarding_steps_count))
+        .check(matches(isDisplayed()))
+    }
+  }
+
+  @RunOn(TestPlatform.ROBOLECTRIC)
+  @Config(qualifiers = "land")
+  @Test
+  fun testFragment_startInLandscapeMode_studentNavigationCardClicked_launchesNewProfileScreen() {
+    launchOnboardingProfileTypeActivity().use {
+      onView(withId(R.id.profile_type_learner_navigation_card)).perform(click())
+      testCoroutineDispatchers.runCurrent()
+      // Does nothing for now, but should fail once navigation is implemented in a future PR.
+      onView(withId(R.id.profile_type_learner_navigation_card))
+        .check(matches(isDisplayed()))
+      onView(withId(R.id.profile_type_learner_navigation_card))
+        .check(
+          matches(
+            hasDescendant(
+              withText(R.string.onboarding_profile_type_activity_student_text)
+            )
+          )
+        )
+      onView(withId(R.id.profile_type_supervisor_navigation_card))
+        .check(matches(isDisplayed()))
+      onView(withId(R.id.profile_type_supervisor_navigation_card))
+        .check(
+          matches(
+            hasDescendant(
+              withText(R.string.onboarding_profile_type_activity_parent_text)
+            )
+          )
+        )
+    }
+  }
+
+  @RunOn(TestPlatform.ESPRESSO)
+  @Test
+  fun testFragment_orientationChange_studentNavigationCardClicked_launchesNewProfileScreen() {
+    launchOnboardingProfileTypeActivity().use {
+      onView(isRoot()).perform(orientationLandscape())
+      testCoroutineDispatchers.runCurrent()
+      // Does nothing for now, but should fail once navigation is implemented in a future PR.
+      onView(withId(R.id.profile_type_learner_navigation_card))
+        .check(matches(isDisplayed()))
+      onView(withId(R.id.profile_type_learner_navigation_card))
+        .check(
+          matches(
+            hasDescendant(
+              withText(R.string.onboarding_profile_type_activity_student_text)
+            )
+          )
+        )
+      onView(withId(R.id.profile_type_supervisor_navigation_card))
+        .check(matches(isDisplayed()))
+      onView(withId(R.id.profile_type_supervisor_navigation_card))
+        .check(
+          matches(
+            hasDescendant(
+              withText(R.string.onboarding_profile_type_activity_parent_text)
+            )
+          )
+        )
+    }
+  }
+
+  @Test
+  fun testFragment_supervisorNavigationCardClicked_launchesProfileChooserScreen() {
+    launchOnboardingProfileTypeActivity().use {
+      onView(withId(R.id.profile_type_supervisor_navigation_card)).perform(click())
+      testCoroutineDispatchers.runCurrent()
+      intended(hasComponent(ProfileChooserActivity::class.java.name))
+    }
+  }
+
+  @RunOn(TestPlatform.ROBOLECTRIC)
```

Comment:
It confuses me when previous comments in Espresso tests say "// Robolectric is usually not used to test the interaction of
  // Android components"
  To me perform(click()) seems like an interaction

Reply:
By android components I meant things like backpress/back button. I realized these are not very good tests, because we don't typically test the android lifecycle(here, I wanted to know wether an activity will finish and go back to the previous one). I have therefore removed these tests.